smart-contracts/contracts/mixins/MixinKeys.sol

```diff
@@ -31,11 +31,21 @@ contract MixinKeys is
 
   event KeyManagerChanged(uint indexed _tokenId, address indexed _newManager);
 
-  // Keys
-  // Each owner can have at most exactly one key
-  // return 0 values when missing a key
+  // DEPREC: dont use
   mapping (address => Key) internal keyByOwner;
 
+  // store all keys: tokenId => token
+  mapping(uint256 => Key) private _keys;
+  
+  // store ownership: owner => list of tokenIds
+  mapping(address => mapping(uint256 => uint256)) private _ownedKeyIds;
+  
+  // store indexes: owner => list of tokenIds
+  mapping(uint256 => uint256) private _ownedKeysIndex;
```

Comment:
needed?

Reply:
yes this is need to keep track of where a token is in array so when a token is removed, we can swap the last token at its index and we don't end up with an empty value in the array